### ISSUE TITLE
Filtering (WHERE) by context value from LET block not working when an index is also present in the WHERE clause.

### ISSUE THREAD

**Issue by misu200** [orientechnologies/orientdb#8572]
OrientDB Version: 3.0.8
Java Version: java version "1.8.0_171"
OS: Windows 10 Home

I ve created the following simple database test:
```

create class A extends V;	
create property A.name  string;
create index A.nameHash ON A (name) NOTUNIQUE_HASH_INDEX;
	   
create class B extends V;
create class X extends E;							
	  
let $a = create vertex A set name="Mike";
let $b = create vertex B;
create edge X from $a to $b;
```

The following query returns the expected non-empty result:
`select $t,$t.size(),* from A let $t=(select * from (select expand($parent.$parent.$current.out()))) where name="Mike" `

Add " and $t.size()>0" at the end and the query will return empty although $t.size() is 1 in our case:

`select $t,$t.size(),* from A let $t=(select * from (select expand($parent.$parent.$current.out()))) where name="Mike" and $t.size()>0`

If i drop the index A.nameHash the above query returns the expected non-empty result.

Thanks,
Mihai

**Comment by luigidellaquila** [comment]
hi @misu200 

Which OrientDB version are you using?

Thanks

Luigi

**Comment by misu200** [comment]
> hi @misu200
> 
> Which OrientDB version are you using?
> 
> Thanks
> 
> Luigi
OrientDB 3.0.8
Sorry for forgetting to add this info to the issue. I ve updated the original post also to include this.

**Comment by luigidellaquila** [comment]
Hi @misu200 

Thank you very much, I'll check it asap

Luigi

**Comment by luigidellaquila** [comment]
Hi @misu200 

I pushed a fix for this problem, it will be released with OrientDB v 3.0.9

Thanks

Luigi
